homework/index.js

```diff
@@ -29,18 +14,99 @@
+    const tBody = createAndAppend('tbody', table);
+    const details = ['Repository:', 'Description:', 'Forks:', 'Updated:'];
+    details.forEach(detail => {
+      const tr = createAndAppend('tr', tBody);
+      createAndAppend('td', tr, { class: 'label', text: detail });
+      createAndAppend('td', tr, { id: detail });
+    });
+    const secondTd = document.getElementById('Repository:');
+    const link = createAndAppend('a', secondTd, {
+      href: repository.html_url,
+      target: '_blank',
+    });
+    link.innerText = repository.name;
+    document.getElementById('Description:').innerText = repository.description;
+    document.getElementById('Forks:').innerText = repository.forks;
+    document.getElementById('Updated:').innerText = new Date(
+      repository.updated_at,
+    ).toLocaleDateString();
+  }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
+  function createContributorsSide(contributorContainer, contributors) {
+    contributorContainer.innerHTML = '';
+    createAndAppend('p', contributorContainer, { text: 'Contributions: ' });
+    contributors.forEach(contributor => {
+      const contributorInfo = createAndAppend('a', contributorContainer, {
+        href: contributor.html_url,
+        target: '_blank',
+      });
+      const contributorDiv = createAndAppend('div', contributorInfo, {
+        class: 'contributor',
+      });
+      createAndAppend('img', contributorDiv, { src: contributor.avatar_url });
+      createAndAppend('p', contributorDiv, { text: contributor.login });
+      createAndAppend('div', contributorDiv, {
+        class: 'contributor-badge',
+        text: contributor.contributions,
+      });
     });
   }
 
+  function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('div', root, { class: 'header' });
+    createAndAppend('p', header, { text: 'HYF Repositories' });
+
+    fetch(url)
+      .then(Response => Response.json())
+      .then(repositories => {
+        const container = createAndAppend('div', root, { id: 'container' });
+        const repoContainer = createAndAppend('div', container, {
+          class: 'left-div',
+        });
+        const contributorContainer = createAndAppend('div', container, {
+          class: 'right-div',
+        });
+        const select = createAndAppend('select', header, { class: 'repo-selector' });
+        repositories.sort((a, b) => a.name.localeCompare(b.name));
+        repositories.forEach(repository => {
+          createAndAppend('option', select, { text: repository.name });
+        });
+        createRepositoryDescription(repoContainer, repositories[0]);
+        fetch(repositories[0].contributors_url)
```

Comment:
In addition to my comment regarding moving this logic to a function, if you remember from our class last Sunday, we talked about promise chaining instead of having nested promises. The way you are trying to fetch the contributors details for the first repository, it is being done using nested promises. Can you try to make use of promise chaining here instead of nested promises?

Something like this inside your `main` function:
```
// (fetch ALL DATA)
fetch(url)
  .then(resp => resp.json())
  .then(data => {
    // create HTML as you are doing already
   // add the select event listener here
  // possibly return something here for the next .then block to use ;)
  })
  .then(/* fetch contributors details here by calling function created after above comment */)
  .catch(/* this need not be touched as it is being done correctly */);
```